File: Algorithm/Emage.java
package jaddu;

import java.awt.Color;
import java.awt.image.BufferedImage;
import java.io.File;
import java.io.IOException;
import java.util.ArrayList;

import javax.imageio.ImageIO;

public class Emage {
	
	public static final int TOP_MARGIN = 13;
	public static final int SPACE = 14;
	static int width;
	static String fileName;

	static ArrayList<Character> chars;
	static ArrayList<Integer> asciiChars;
	static BufferedImage originalImage;
	static File file = null;

	
	public static void main(String[] args) throws IOException {
		fileName=new String("output.png");
		String text=new String("Hello");
		if(text.length()<10)
			text= "0"+text.length() + text;
		else
			text= text.length() + text;
		
		chars = new ArrayList<Character>();
		asciiChars = new ArrayList<Integer>();
		originalImage = ImageIO.read(new File(fileName));

		
		width=originalImage.getWidth();
		encryptionImg(text);
	}
	
	public static void encryptionImg(String text) {
		int index = 0;

		//for each loop to add characters
		for (char c : text.toCharArray())
			chars.add(c);

		//loop to store ascii value of char
		for (char c : chars)
			asciiChars.add((int) c);

		
		for (int x = 0; x < width; x++) {
			if (x % SPACE == 0 && index < asciiChars.size()) {
				Color greenShade = new Color(1, asciiChars.get(index), 1);
				originalImage.setRGB(x, TOP_MARGIN, greenShade.getRGB());
				index++;
			}
		}

		try {

			file = new File("outputfile.png");
			ImageIO.write(originalImage, "png", file);
			System.out.println("done");
		} catch (IOException e) {
			System.out.println("Error: " + e);
		}
	}

	public int getImageWidth() {
		return this.width;
	}



}

File: Algorithm/Emage.py
from PIL import Image
MARGIN=13
SPACE=14
##pass image using Image.open
##input
im=Image.open("D:\workspace\Image-Encrypt-Decrypt\output.png")
pix=im.load()
##get width of the image
width=im.size[0]
text="helloworldimjaddu"
lenText=len(text)
if(lenText<10):
    text="0"+str(lenText)+text
else:
    text=str(lenText)+text
asciiChars=[ord(c) for c in text]
index=0
##encrytion process
x=0
while x<width:
    if(x%SPACE==0 and index<len(asciiChars)):
        pix[x,MARGIN]=(1,asciiChars[index],1)
        index+=1
    x+=1

##save image
##output
im.save("D:\workspace\Image-Encrypt-Decrypt\outputpython.png")


##finish
